src/regmod/utils.py

```diff
@@ -21,12 +45,62 @@ def default_vec_factory(vec: Any, size: int, default_value: float,
     return vec
 
 
-def check_size(vec: Any, size: int, vec_name: str = 'vec'):
+def check_size(vec: np.ndarray, size: int, vec_name: str = 'vec') -> None:
+    """Check the size of the vector.
+
+    Parameters
+    ----------
+    vec : np.ndarray
+        Vector to be validated.
+    size : int
+        The assumption size of the vector.
+    vec_name : str, optional
+        Name of the vector, used for print error purpose. Default to ``'vec'``.
+
+    Raises
+    ------
+    AssertionError
+        Raised when the size of the vector not match with the given size.
+    """
     assert len(vec) == size, f"{vec_name} must length {size}."
 
 
 @dataclass
 class SplineSpecs:
+    """Spline parameters used to create spline.
+
+    Attributes
+    ----------
+    knots : np.ndarray
+        Knots placement of the spline. Depends on ``knots_type`` this will be
+        used differently.
+    degree : int, optional
+        Degree of the spline. Default to be 3.
+    l_linear : bool, optional
+        If ``True``, spline will use left linear tail. Default to be ``False``.
+    r_linear : bool, optional
+        If ``True``, spline will use right linear tail. Default to be `False``.
+    include_first_basis : bool, optional
+        If ``True``, spline will include the first basis of the spline. Default
+        to be ``True``.
+    knots_type : str, optional
```

Comment:
Sometimes I've seen the options listed rather than the type, with the default first. For example:

`knots_type : {'abs', 'rel_domain', 'rel_freq'}, optional`

Reply:
Great! thanks very much! I like it! Here is the reference from `numpydoc`.
https://numpydoc.readthedocs.io/en/latest/format.html#parameters